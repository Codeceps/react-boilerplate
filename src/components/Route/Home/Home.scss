@import '~resources/css/Variables';

@mixin font-align($font-name) {
    font-family: $font-name;
    text-align: center;
}

#home-title {
    @include font-align($font-pacifico);
    color: chocolate;
}