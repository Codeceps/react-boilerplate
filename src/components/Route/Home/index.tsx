import React from "react";

import './Home.scss';


export interface HomeProps {
    initialText: string
}

export interface HomeState {
    inputText: string
    storedText: string
}

export default class Home extends React.Component<HomeProps, HomeState> {

    state: HomeState = {
        inputText: '',
        storedText: this.props.initialText
    }

    constructor(props: HomeProps) {
        super(props);
    }

    onChangeListener = (event: React.ChangeEvent<HTMLInputElement>) => {
        this.setState({
            inputText: event.target.value
        });
    }

    onSubmitListener = (event: React.FormEvent<HTMLFormElement>) => {
        event.preventDefault();
        this.setState({
            storedText: this.state.inputText ? this.state.inputText : this.props.initialText
        });
    }


    render() {
        const {inputText, storedText} = this.state;

        return (
            <div>    
                <h2 id='home-title'>Stored text: {storedText}</h2>

                <form onSubmit={this.onSubmitListener}>
                    <input 
                        className='centered-widget' 
                        value={inputText}
                        type="text" 
                        placeholder='Enter text' 
                        onChange={this.onChangeListener}
                    />

                    <input 
                        className='centered-widget'
                        type="submit" 
                        value='Store text'
                    />
                </form>

            </div>
        )
    }

}