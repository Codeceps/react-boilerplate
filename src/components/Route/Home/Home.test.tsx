import React from "react";
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Home from './';


Enzyme.configure({adapter: new Adapter()});

describe('<Home />', () => {

    it('should render without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Home initialText=''/>, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    /**
     * Snapshot test using 'react-test-renderer'
     */
    it('should have valid snapshot', () => {
        const component = renderer.create(<Home initialText=''/>)
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    /**
     * DOM test using 'enzyme'
     */
    it('should update the header on form submit', () => {
        const component = shallow(<Home initialText='' />);

        component.find('input[type="text"]').simulate('change', { target: { value: 'New Value' } });
        component.find('form').simulate('submit', { preventDefault: () => {} });
        
        let headerText = component.find('h2').text();
        expect(headerText).toBe('Stored text: New Value')
    });

});