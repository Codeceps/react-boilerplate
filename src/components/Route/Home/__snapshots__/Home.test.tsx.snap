// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should have valid snapshot 1`] = `
<div>
  <h2
    id="home-title"
  >
    Stored text: 
    
  </h2>
  <form
    onSubmit={[Function]}
  >
    <input
      className="centered-widget"
      onChange={[Function]}
      placeholder="Enter text"
      type="text"
      value=""
    />
    <input
      className="centered-widget"
      type="submit"
      value="Store text"
    />
  </form>
</div>
`;
